src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movieSession " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getMovieSessionQuery = session.createQuery(
+                    "from MovieSession ms left join fetch ms.movie where ms.id = :id",
+                    MovieSession.class
+            );
+            getMovieSessionQuery.setParameter("id", id);
+            return getMovieSessionQuery.getSingleResult();
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getMovieSessionQuery = session.createQuery(
+                    "from MovieSession ms left join fetch ms.movie where ms.showTime = :date",
+                    MovieSession.class
+            );
+            getMovieSessionQuery.setParameter("date", date);
+            return getMovieSessionQuery.getResultList();

```

Comment:
In the `findAvailableSessions()` method, you are getting all `MovieSession` from dao and then sort them on service layer. This is not good for performance. It's better to extract 'ready to use' data from DB on dao layer. Also, movie sessions should have running time limited within the day that your receive as input parameter: `LocalDate date`.